components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java

```diff
@@ -1074,4 +1096,70 @@ public synchronized boolean sendSiteCommand(String command) throws GenericFileOp
         // is not implemented
         return true;
     }
+
+    /*
+     * adapted from com.jcraft.jsch.Util.createSocket(String, int, int)
+     *
+     * added possibility to specify the address of the local network interface, against the
+     * connection should bind
+     */
+    static Socket createSocketUtil(String host, int port, String bindAddress, int timeout) {
+        Socket socket=null;
+        if(timeout==0){
+          try{
+            socket=bindAddress == null ? new Socket(host, port) : new Socket(InetAddress.getByName(host), port, InetAddress.getByName(bindAddress), 0);
+            return socket;
+          }
+          catch(Exception e){
+            String message=e.toString();
+            if(e instanceof Throwable)
+              throw new RuntimeException(message, (Throwable)e);
+            throw new RuntimeException(message);
+          }
+        }
+        final String _host=host;
+        final int _port=port;
+        final Socket[] sockp=new Socket[1];
+        final Exception[] ee=new Exception[1];
+        String message="";
+        Thread tmp=new Thread(new Runnable(){
+            public void run(){
+              sockp[0]=null;
+              try{
+                sockp[0]=bindAddress == null ? new Socket(_host, _port) : new Socket(InetAddress.getByName(_host), _port, InetAddress.getByName(bindAddress), 0);
+              }
+              catch(Exception e){
+                ee[0]=e;
+                if(sockp[0]!=null && sockp[0].isConnected()){
+                  try{
+                    sockp[0].close();
+                  }
+                  catch(Exception eee){}
+                }
+                sockp[0]=null;
+              }
+            }
+          });
+        tmp.setName("Opening Socket "+host);
+        tmp.start();
+        try{
+          tmp.join(timeout);
+          message="timeout: ";
+        }
+        catch(java.lang.InterruptedException eee){
+        }
+        if(sockp[0]!=null && sockp[0].isConnected()){
+          socket=sockp[0];
+        }
+        else{
+          message+="socket is not established";
+          if(ee[0]!=null){
+            message=ee[0].toString();
+          }
+          tmp.interrupt();
+          tmp=null;
+          throw new RuntimeException(message, ee[0]);
```

Comment:
RuntimeCamelException would be bettter..

Reply:
Thanks for the review, I've changed this accordingly.